sunpy/io/_jp2.py

```diff
@@ -176,7 +176,7 @@ def write(fname, data, header, **kwargs):
 
     tmpname = fname + "tmp.jp2"
     jp2_data = np.uint8(data)
-    jp2 = Jp2k(tmpname, jp2_data, **kwargs)
+    jp2 = Jp2k(tmpname, np.flip(jp2_data, 0), **kwargs)
```

Comment:
Add a comment that explains that we have to flip the array back here because we flip it upon reading (due to the difference in the origin of the data)

Reply:
Where does it get flipped when its read in?

Reply:
I added a comment here, and updated the test to verify the data array after reading it back.